### ISSUE TITLE
False positive `invalid-argument-type` for a decorator with arguments

### ISSUE THREAD

**Issue by Yura52** [astral-sh/ty#3192]
### Summary

(I could not quickly understand if the issue is the same as #1701, so opened a new one just in case)

In the following code, ty marks all three `return` expression with `invalid-argument-type`. Software versions:

- Python: 3.13.12
- ty: 0.0.27

```python
import functools
from collections.abc import Callable


def augment_output[**P, T](
    value: int,
) -> Callable[[Callable[P, T]], Callable[P, tuple[T, int]]]:
    def decorator(fn: Callable[P, T]) -> Callable[P, tuple[T, int]]:
        @functools.wraps(fn)
        def wrapper(*args: P.args, **kwargs: P.kwargs) -> tuple[T, int]:
            return fn(*args, **kwargs), value  # <-- invalid-return-type + invalid-argument-type x2

        return wrapper  # <-- invalid-return-type

    return decorator  # <-- invalid-return-type


@augment_output(1)
def f():
    return 0


print(f())
# prints:
# (0, 1)
```

**Comment by Yura52** [comment]
Oh, and it turns out that in this line:

```
return fn(*args, **kwargs), value
```

ty also marks `args` and `kwargs` with `invalid-argument-type`

**Comment by carljm** [comment]
Thanks! This looks like #2723.
